release/e2e.py

```diff
@@ -477,67 +478,83 @@ def report_result(test_suite: str, test_name: str, status: str, logs: str,
         f"results, artifacts, category) "
         f"VALUES (:created_on, :test_suite, :test_name, :status, :last_logs, "
         f":results, :artifacts, :category)")
+    parameters = [{
+        "name": "created_on",
+        "typeHint": "TIMESTAMP",
+        "value": {
+            "stringValue": now.strftime("%Y-%m-%d %H:%M:%S")
+        },
+    }, {
+        "name": "test_suite",
+        "value": {
+            "stringValue": test_suite
+        }
+    }, {
+        "name": "test_name",
+        "value": {
+            "stringValue": test_name
+        }
+    }, {
+        "name": "status",
+        "value": {
+            "stringValue": status
+        }
+    }, {
+        "name": "last_logs",
+        "value": {
+            "stringValue": logs
+        }
+    }, {
+        "name": "results",
+        "typeHint": "JSON",
+        "value": {
+            "stringValue": json.dumps(results)
+        },
+    }, {
+        "name": "artifacts",
+        "typeHint": "JSON",
+        "value": {
+            "stringValue": json.dumps(artifacts)
+        },
+    }, {
+        "name": "category",
+        "value": {
+            "stringValue": category
+        }
+    }]
+
+    retry_cnt = 0
+    # Default boto3 call timeout is 45 seconds.
+    retry_delay_s = 64
+    MAX_RDS_RETRY = 3
+    db_updated = False
+
+    while retry_cnt < MAX_RDS_RETRY and not db_updated:
```

Comment:
It's unfortunate that boto3 won't auto retry StatementTimeoutException based on https://boto3.amazonaws.com/v1/documentation/api/latest/guide/retries.html.

Do you think we can put the retry logic into a decorator so we can not only retry this but also anyscale sdk calls.

Reply:
Do we have any anyscale sdk call we should retry? I think instead of decorator we should write a helper function like

```python3
def exponential_backoff_retry(lambda, on=[list of exception to catch], max_retry_count)
```

Wdyt?

Reply:
Yes, that works. We should at least retry `result = sdk.get_session_command(session_command_id=scd_id)`.

Reply:
Actually let me merge this first. It's because it takes some time to test this change (I think we should start writing unit tests... This code is getting more complicated cc @krfricke), and I have some other tasks I want to finish before doing this. I will do this refactoring + retrying the session command in the beg of next sprint

Reply:
Added to my next sprint backlog